### PR TITLE
Remove types from comments.

### PR DESCRIPTION
**Pull request by clokep**
These always annoy me. I came up with a regex to find them:

```bash
$ grep -rE "^ +[a-zA-Z_]+ \(.+\) *[:$]" synapse tests | grep -vE ": +(if|for|except) "
$ grep -rE -A1 "^ +Returns?:?$" synapse tests | grep -E "\- +[a-zA-Z].+:"
```

This doesn't clean-up all instance, namely it doesn't touch ones that refer to class args. Those seemed somewhat useful still? At least in some cases.

It also adds a few missing types where it seemed easier to just fill them in from the comments.

### CODE REVIEW

**Comment by clokep** [comment]
Oh hmm old deps is failing here. Will need to investigate.

File: tests/replication/slave/storage/test_events.py [hunk 1]
```diff
@@ -280,6 +282,7 @@ def test_get_rooms_for_user_with_stream_ordering_with_multi_event_persist(self):
         j2, j2ctx = self.build_event(
             type="m.room.member", sender=USER_ID_2, key=USER_ID_2, membership="join"
         )
+        assert j2.internal_metadata.stream_ordering is not None
```
[the remark this answers is not in the document]
Comment (clokep) [reply tests/replication/slave/storage/test_events.py:285]: Without this, we get a mypy error. I think I might have put the assertion too early though. 👍

File: synapse/handlers/e2e_room_keys.py [hunk 2]
```diff
@@ -377,8 +377,8 @@ async def delete_version(self, user_id: str, version: Optional[str] = None) -> N
         """Deletes a given version of the user's e2e_room_keys backup
 
         Args:
-            user_id(str): the user whose current backup version we're deleting
-            version(str): the version id of the backup being deleted
+            user_id: the user whose current backup version we're deleting
+            version: the version id of the backup being deleted
```
Comment (DMRobertson) [inline synapse/handlers/e2e_room_keys.py:381]: What does `version=None` mean here---looks like it means "the most recent backup"?

(FWIW my guess would have been "delete all versions")

Comment (clokep) [reply synapse/handlers/e2e_room_keys.py:381]: This copies the docstring from `delete_e2e_room_keys_version`:

```suggestion
            version: Optional. the version ID of the backup version we're deleting
                If missing, we delete the current backup version info.
```

File: tests/unittest.py [hunk 3]
```diff
@@ -360,13 +360,13 @@ def wait_for_background_updates(self) -> None:
                 store.db_pool.updates.do_next_background_update(False), by=0.1
             )
 
-    def make_homeserver(self, reactor, clock):
+    def make_homeserver(self, reactor: MemoryReactor, clock: Clock):
```
[the remark this answers is not in the document]
Comment (clokep) [reply tests/unittest.py:363]: This makes a lot of things fail (I had tried it before putting up this PR). IMO it is not worth holding up this PR to add that.

File: synapse/server_notices/consent_server_notices.py [hunk 4]
```diff
@@ -113,9 +113,8 @@ def copy_with_str_subst(x: Any, substitutions: Any) -> Any:
     """Deep-copy a structure, carrying out string substitutions on any strings
 
     Args:
-        x (object): structure to be copied
-        substitutions (object): substitutions to be made - passed into the
-            string '%' operator
+        x: structure to be copied
+        substitutions: substitutions to be made - passed into the string '%' operator
```
[the remark this answers is not in the document]
Comment (clokep) [reply synapse/server_notices/consent_server_notices.py:117]: Maybe, but my goal in this PR wasn't to get rid of `Any`.

File: synapse/storage/databases/main/e2e_room_keys.py [hunk 5]
```diff
@@ -391,10 +391,10 @@ async def get_e2e_room_keys_version_info(
...
             A dict giving the info metadata for this backup version, with
             fields including:
-                version(str)
-                algorithm(str)
-                auth_data(object): opaque dict supplied by the client
-                etag(int): tag of the keys in the backup
+                version (str)
+                algorithm (str)
+                auth_data (object): opaque dict supplied by the client
+                etag (int): tag of the keys in the backup
```
[the remark this answers is not in the document]
Comment (DMRobertson) [reply synapse/storage/databases/main/e2e_room_keys.py:397]: Oh I see. Thanks.

File: synapse/storage/databases/main/room.py [hunk 6]
```diff
@@ -2057,7 +2057,7 @@ async def get_event_report(self, report_id: int) -> Optional[Dict[str, Any]]:
         Args:
             report_id: ID of reported event in database
         Returns:
-            event_report: json list of information from event report
+            json list of information from event report
```
[the remark this answers is not in the document]
Comment (clokep) [reply synapse/storage/databases/main/room.py:2060]: I think so? Not sure why this said `json list`...

```suggestion
            JSON dict of information from an event report or None if it
            does not exist.
```

File: synapse/handlers/saml.py [hunk 7]
```diff
@@ -441,7 +441,7 @@ def saml_response_to_user_attributes(
             client_redirect_url: where the client wants to redirect to
 
         Returns:
-            dict: A dict containing new user attributes. Possible keys:
+            A dict containing new user attributes. Possible keys:
                 * mxid_localpart (str): Required. The localpart of the user's mxid
                 * displayname (str): The displayname of the user
                 * emails (list[str]): Any emails for the user
```
[the remark this answers is not in the document]
Comment (clokep) [reply synapse/handlers/saml.py:447]: Agreed, but I'm going to leave that to the future.

**Review by DMRobertson — APPROVED** [review]

**Comment by clokep** [comment]
Thanks for the review! 🎉

**Pull request merged**